src/NUnitFramework/framework/Constraints/CollectionOrderedConstraint.cs

```diff
@@ -103,7 +109,55 @@ public CollectionOrderedConstraint Using<T>(Comparison<T> comparer)
         /// </summary>
         public CollectionOrderedConstraint By(string propertyName)
         {
-            this.propertyName = propertyName;
+            if (this.properties != null && this.properties.Count != 0)
+                throw new InvalidOperationException("Cannot use By on existing collection. Use ThenBy to add additional items.");
```

Comment:
It can be allowed, but we must be careful to only take the status of the `Descending` property into account if there's no item in the collection yet.

If someone would call `Is.Ordered.Descending.By("SomeProp").By("AnotherProp")` the descending status should only be set to the first property and discarded for all subsequent calls.